test/rpc_test.py

```diff
@@ -827,3 +827,19 @@ def test_remote_same_worker(self):
             "worker{}".format(dst_rank), my_rref_function, args=(rref_a, rref_b)
         )
         self.assertEqual(rref_c.to_here(), torch.ones(n, n) + 4)
+
+    def test_sender_exceptions(self):
+        dist.init_process_group(backend="gloo", init_method=self.init_method, timeout=timedelta(seconds=5))
+        rpc.init_model_parallel(
+            self_name="worker%d" % self.rank,
+            backend=TEST_CONFIG.backend,
+            self_rank=self.rank,
+            init_method=self.init_method,
+        )
+
+        if self.rank == 0:
+            # allow worker 1 to exit without joining
+            time.sleep(0.5)
```

Comment:
@pritamdamania87 , `dist.barrier()` doesn't work here since the `connection reset by peer` error gets raised, because worker 1 exits without joining (but this is required for the test). So I reverted it back to time.sleep. We could also do
```
try:
    dist.barrier()
except RuntimeError:
   pass
```

but this seems even hackier than time.sleep.

Reply:
Maybe increase the duration to 3s or so to be conservative since sometimes with asan/tsan tests run really slowly.